encoder/vaapiencoder_vp9.cpp

```diff
@@ -0,0 +1,323 @@
+/*
+ * Copyright (C) 2014 Intel Corporation. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
+#include "vaapiencoder_vp9.h"
+#include "common/scopedlogger.h"
+#include "common/common_def.h"
+#include "vaapi/vaapicontext.h"
+#include "vaapi/vaapidisplay.h"
+#include "vaapicodedbuffer.h"
+#include "vaapiencpicture.h"
+#include "vaapiencoder_factory.h"
+#include <algorithm>
+
+namespace YamiMediaCodec {
+
+// each frame can have at most 3 reference frames, golden, alt and last
+enum maxSizeValues {
+    kMaxReferenceFrames = 8,
```

Comment:
"each frame can have at most 3 reference frames, golden, alt and last", so why kMaxReferenceFrames is 8, not 3?